### ISSUE TITLE
DNS rewrite does not working

### ISSUE THREAD

**Issue by ericxliu1990** [AdguardTeam/AdGuardHome#4170]
Have a question or an idea? Please search it [on our forum](https://github.com/AdguardTeam/AdGuardHome/discussions) to make sure it was not yet asked. If you cannot find what you had in mind, please [submit it here](https://github.com/AdguardTeam/AdGuardHome/discussions/new).

### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x ] I am running the latest version
- [x] I checked the documentation and found no answer
- [x ] I checked to make sure that this issue has not already been filed

### Issue Details

* **Version of AdGuard Home server:**
  * v0.107.2
* **How did you install AdGuard Home:**
  * GitHub releases
* **How did you setup DNS configuration:**
  * Router
* **If it's a router or IoT, please write device model:**
  * openwrt on Linksys WRT3200ACM
* **CPU architecture:**
  * Armv7
* **Operating system and version:**
  * openwort OpenWrt 19.07.6

### Expected Behavior
Adguard rewrites the DNS request to a preset IP address
### Actual Behavior

Adguard does no rewrite the DNS request directly but to resolve the DNS upstream. 
### Screenshots

<details><summary>Screenshot:</summary>

![image](https://user-images.githubusercontent.com/3407412/150286498-bf713aca-46df-4edb-8c0c-e5ac19c4e729.png)

![image](https://user-images.githubusercontent.com/3407412/150286598-0d60901d-309b-46a2-b26b-e738ea68395e.png)

![image](https://user-images.githubusercontent.com/3407412/150286693-25a44613-f945-4cd6-bb1c-f0d51d1e2c62.png)

</details>

### Additional Information
$nslookup diskstation.local 10.10.0.1
Server:		10.10.0.1
Address:	10.10.0.1#53

** server can't find diskstation.local: NXDOMAIN

dig diskstation.local

; <<>> DiG 9.10.6 <<>> diskstation.local
;; global options: +cmd
;; Got answer:
;; WARNING: .local is reserved for Multicast DNS
;; You are currently testing what happens when an mDNS query is leaked to DNS
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 62198
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;diskstation.local.		IN	A

;; AUTHORITY SECTION:
.			875	IN	SOA	a.root-servers.net. nstld.verisign-grs.com. 2022012000 1800 900 604800 86400

;; Query time: 81 msec
;; SERVER: 10.10.0.1#53(10.10.0.1)
;; WHEN: Wed Jan 19 22:40:48 PST 2022
;; MSG SIZE  rcvd: 121

**Comment by agneevX** [comment]
What happens if you delete the second incorrect rewrite rule?

**Comment by ericxliu1990** [comment]
Thanks Ainar for the quick response. You are right. Checked “Block domains using filters and hosts files” resolves this issue. 

In my case, I have home assistant integration with Adguard. The setting was incorrectly changed due to an incorrect automation (close all device when I left home and this Adguard setting is a "device" in the home). 

Re agneevX
I think the second rule is a valid CNAME rewrite. I used it for reverse proxy resolution.

Re  Stooovie
I don't believe so. I use the DHCP server from Openwrt.
